euphony/src/main/java/co/euphony/rx/EuRxManager.java

```diff
@@ -96,14 +107,24 @@ private boolean listenOnNative() {
 	}
 
 	public boolean listen() {
-		if(rxEngineType == RxEngineType.EUPHONY_JAVA_ENGINE)
-			return listenOnJava();
+		return listen(0, () -> {
+		});
+	}
+
+	public boolean listen(@IntRange(from = 0) long timeout) {
+		return listen(timeout, () -> {
+		});
```

Comment:
Same comments with above :)
```suggestion
		return listen(0, null);
```